### ISSUE TITLE
POST label to bucket with invalid labelID returns unparsable JSON response body

### ISSUE THREAD

**Issue by karel-rehor** [influxdata/influxdb#23419]
I am filling in tests for Grace.  The following error handling issue arose when trying to parse the response through supertest.  

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

In Postman or similar

1. Create a bucket
2. Create a label
3. POST the new label into the bucket endpoint `api/v2/buckets/c1651eb0ab21cb07/labels`, however in the request body shorten the actual labelID. e.g. `{ labelID: '097de5a3' }`

__Expected behavior:__
Up to know when working with Supertest all promises returned from requests, whether the response status is in the 200 range or 400 and 500 range, have resolved with a response object, with its contents unmarshalled using `JSON.parse()` - in the supertest library. 

So the response body returned should be parsable using the javascript JSON object. 

Furthermore, from writing similar tests with invalid IDs, I expected a similar error message: "id must have length of 16 bytes".

__Actual behavior:__
The supertest promise finishes with "reject" and error.  `SyntaxError: Unexpected token { in JSON at position 68`

Inspection through Postman reveals that two error objects are being returned in the response body. 

```
POST https://twodotoh-dev-bonitoo-kk.remocal.influxdev.co/api/v2/buckets/c1651eb0ab21cb07/labels
400
337 ms
POST /api/v2/buckets/c1651eb0ab21cb07/labels HTTP/1.1
Authorization: Token I3uHfzZWdKLKZgE4g-ZzeoWGT2B2yM-y_j0cGjsHKnrnHTCTo6a2GCfha28XPc_dmhAdKYLDQWytp0Z0oISWMA==
Content-Type: application/json
User-Agent: PostmanRuntime/7.29.0
Accept: */*
Postman-Token: 24ba291a-6fd5-492a-b32a-0a1f8db571de
Host: twodotoh-dev-bonitoo-kk.remocal.influxdev.co
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Content-Length: 29
 
{
"labelID": "097de5a3"
}
 
HTTP/1.1 400 Bad Request
Date: Thu, 09 Jun 2022 08:37:01 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 126
Connection: keep-alive
Trace-Id: 224480c511f58da7
Trace-Sampled: false
X-Platform-Error-Code: invalid
Strict-Transport-Security: max-age=15724800; includeSubDomains
X-Influxdb-Request-ID: 000c00c01e002143da0a890964594097
X-Influxdb-Build: Cloud
 
{
"code": "invalid",
"message": "Invalid post label map request"
}{
"code": "invalid",
"message": "label id is required"
}
``` 

__Environment info:__

Testing against K8SIDPE remocal 

```
commit 3a6bbd708a14cef2b14fde6e39f8bea8446750fe (HEAD -> master, origin/master, origin/alchemy/automatic, origin/HEAD)
Author: Hercules Mango Churchill <alchemy@influxdata.com>
Date:   Thu Jun 9 07:49:16 2022 +0000
```

__Response from other error - which can be handled__

For comparison with response body above.  

```
POST https://twodotoh-dev-bonitoo-kk.remocal.influxdev.co/api/v2/buckets/c1651eb0ab21cb07/labels
400
328 ms
POST /api/v2/buckets/c1651eb0ab21cb07/labels HTTP/1.1
Authorization: Token I3uHfzZWdKLKZgE4g-ZzeoWGT2B2yM-y_j0cGjsHKnrnHTCTo6a2GCfha28XPc_dmhAdKYLDQWytp0Z0oISWMA==
Content-Type: application/json
User-Agent: PostmanRuntime/7.29.0
Accept: */*
Postman-Token: b4402eea-663f-4b1b-98bf-b51eba35c65c
Host: twodotoh-dev-bonitoo-kk.remocal.influxdev.co
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Content-Length: 8
 
{
}
 
HTTP/1.1 400 Bad Request
Date: Thu, 09 Jun 2022 08:48:34 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 58
Connection: keep-alive
Trace-Id: 719b2f36c42245c3
Trace-Sampled: false
X-Platform-Error-Code: invalid
Strict-Transport-Security: max-age=15724800; includeSubDomains
X-Influxdb-Request-ID: 8498ad20848103a49d7fc944684a02e0
X-Influxdb-Build: Cloud
 
{
"code": "invalid",
"message": "label id is required"
}

```

**Comment by subho-13** [comment]
Can you please assign me this issue if it's not already fixed?

**Comment by jeffreyssmith2nd** [comment]
I don't believe it has been fixed, you are welcome to take a look at fixing it. Let us know if you have any questions!

**Comment by ghazanhaider** [comment]
I have the same issue with influxdb 2.6.1. 
POST writes to other endpoints work well.
This one, I've tried changing the json payload a few different ways with no luck:

curl -vvvv -X POST -H 'Authorization: Token REDACTED' -H 'Content-Type: application/json' "https://ghazan.dev/api/v2/variables/0a9369424f9d1000/labels" --data '{"LabelID":"ON"}'
Note: Unnecessary use of -X or --request, POST is already inferred.


*   Trying 3.86.212.89:443...
* Connected to ghazan.dev (3.86.212.89) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (OUT), TLS handshake, Client hello (1):
* (304) (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-ECDSA-AES256-GCM-SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=ghazan.dev
*  start date: Jan 10 00:17:25 2023 GMT
*  expire date: Apr 10 00:17:24 2023 GMT
*  subjectAltName: host "ghazan.dev" matched cert's "ghazan.dev"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: POST]
* h2h3 [:path: /api/v2/variables/0a9369424f9d1000/labels]
* h2h3 [:scheme: https]
* h2h3 [:authority: ghazan.dev]
* h2h3 [user-agent: curl/7.85.0]
* h2h3 [accept: */*]
* h2h3 [authorization: Token REDACTED]
* h2h3 [content-type: application/json]
* h2h3 [content-length: 16]
* Using Stream ID: 1 (easy handle 0x7fde4480fc00)
> POST /api/v2/variables/0a9369424f9d1000/labels HTTP/2
> Host: REDACTED
> user-agent: curl/7.85.0
> accept: */*
> authorization: Token REDACTED
> content-type: application/json
> content-length: 16
>
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
* We are completely uploaded and fine
< HTTP/2 400
< server: nginx/1.22.0
< date: Wed, 11 Jan 2023 04:18:54 GMT
< content-type: application/json; charset=utf-8
< content-length: 61
< x-influxdb-build: OSS
< x-influxdb-version: v2.6.1
< x-platform-error-code: invalid
<
* Connection #0 to host influxdb.ghazan.dev left intact
{"code":"invalid","message":"Invalid post label map request"}
